src/server/plugins/engine/pageControllers/FileUploadPageController.ts

```diff
@@ -141,13 +197,28 @@ export class FileUploadPageController extends QuestionPageController {
       const state = await this.getState(request)
       const uploadState = state.upload?.[path] ?? { files: [] }
 
-      // Check for any removed files in the POST payload
-      const removed = await this.checkRemovedFiles(request, uploadState)
+      const { confirm } = this.getFormData(request)
 
-      if (removed) {
+      // Check for any removed files in the POST payload
+      if (confirm === true) {
```

Comment:
Is the === true comparison necessary? Since confirm is already a boolean, the if block would execute if it is truthy

Reply:
Thanks @mokhld, good point

The schema is `Joi.boolean().default(false)` so I'll update **RepeatPageController.ts** too

https://github.com/DEFRA/forms-runner/blob/c6c545ac8d9ae90b4bd57d69580f1a8eca382bf6/src/server/plugins/engine/pageControllers/RepeatPageController.ts#L339

Reply:
Fixed in https://github.com/DEFRA/forms-runner/pull/619/commits/a3741086ae612cedfebc512bc274537e3d676c5c